components/engine_rocks/src/misc.rs

```diff
@@ -126,11 +126,16 @@ impl RocksEngine {
 }
 
 impl MiscExt for RocksEngine {
-    fn flush_cfs(&self, wait: bool) -> Result<()> {
+    fn flush_cfs(&self, cfs: &[&str], wait: bool) -> Result<()> {
         let mut handles = vec![];
-        for cf in self.cf_names() {
+        for cf in cfs {
             handles.push(util::get_cf_handle(self.as_inner(), cf)?);
         }
+        if handles.is_empty() {
+            for cf in self.cf_names() {
+                handles.push(util::get_cf_handle(self.as_inner(), cf)?);
+            }
+        }
```

Comment:
minor: maybe it's cleaner to use a temp cfs to cover both cases.
something like:
   let cfs = cfs.is_empty()? &self.cf_names() : cfs;
   for cf in cfs .....

Reply:
`cf_names` will allocate.

Reply:
`cf_names` will return a `Vec`, so you can put it inside if branch. Instead, it needs to allocate first, and then choose what to borrow.